core/src/main/java/io/grpc/LoadBalancer.java

```diff
@@ -461,10 +462,34 @@ public void updateOobChannelAddresses(ManagedChannel channel, EquivalentAddressG
      * <p>The channel will hold the picker and use it for all RPCs, until {@code updatePicker()} is
      * called again and a new picker replaces the old one.  If {@code updatePicker()} has never been
      * called, the channel will buffer all RPCs until a picker is provided.
+     *
+     * <p>Using this method implies that this load balancer doesn't support channel state, and the
+     * application will get exception when trying to get the channel state.
      */
+    // TODO(zdapeng): add '@deprecated Please migrate ALL usages to {@link #updateBalancingState}'
+    // TODO(zdapeng): and add '@Deprecated'
     public abstract void updatePicker(SubchannelPicker picker);
 
     /**
+     * Set a new state with a new picker to the channel.
+     *
+     * <p>When a new picker is provided via {@code updateBalancingState()}, the channel will apply
+     * the picker on all buffered RPCs, by calling {@link SubchannelPicker#pickSubchannel(
+     * LoadBalancer.PickSubchannelArgs)}.
+     *
+     * <p>The channel will hold the picker and use it for all RPCs, until {@code
```

Comment:
Please also say that the passed state will be the channel's new state, and that `SHUTDOWN` should not be passed and its behavior is undefined.